src/update.rs

```diff
@@ -0,0 +1,305 @@
+// This file contains the functions used for checking for updates and automatically updating the tod CLI tool.
+// Functions that attempt to detect the installation method of the current executable, used for autoupdate and debug
+use std::{env, process::Command};
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum InstallMethod {
+    Homebrew,
+    Scoop,
+    Cargo,
+    FromSource,
+    Unknown,
+}
+
+// Public API: returns the detected or overridden install method if manually specified
+pub fn get_install_method(override_arg: &Option<String>) -> InstallMethod {
+    if let Some(value) = override_arg {
+        match value.trim().to_lowercase().as_str() {
+            "cargo" => InstallMethod::Cargo,
+            "scoop" => InstallMethod::Scoop,
+            "homebrew" => InstallMethod::Homebrew,
+            "source" | "fromsource" => InstallMethod::FromSource,
+            _ => InstallMethod::Unknown,
+        }
+    } else {
+        detect_install_method()
+    }
+}
+
+fn detect_install_method() -> InstallMethod {
+    let path = match env::current_exe() {
+        Ok(p) => p,
+        Err(_) => return InstallMethod::Unknown,
+    };
+
+    let components: Vec<_> = path
+        .components()
+        .map(|c| c.as_os_str().to_string_lossy().to_lowercase())
+        .collect();
+
+    if cfg!(debug_assertions) || components.iter().any(|c| c == "target") {
+        InstallMethod::FromSource
+    } else if components.iter().any(|c| c.contains(".cargo")) {
+        InstallMethod::Cargo
+    } else if components.iter().any(|c| c.contains("scoop")) {
+        InstallMethod::Scoop
+    } else if components.iter().any(|c| c.contains("homebrew")) {
+        InstallMethod::Homebrew
+    } else {
+        InstallMethod::Unknown
+    }
+}
+
+// Public API: returns the string name of how it was installed
+pub fn get_install_method_string(override_arg: &Option<String>) -> &'static str {
```

Comment:
Can we put public API functions at the top of the module when possible? I wonder if we could make a way to lint this....

Just makes it easier for someone reading the module.